Recommended version of gcc-arm-none-eabi for building FW

What is your experience to building project with various versions of gcc-arm-none-eabi?

I tried `gcc-arm-none-eabi-10-2020-q4-major` and have some warnings:
```
usb_cdc.c: In function 'usb_cdc_port_start_tx':
usb_cdc.c:447:41: warning: array subscript '<unknown>' is outside the bounds of an interior zero-length array 'uint8_t[0]' {aka 'unsigned char[]'} [-Wzero-length-bounds]
  447 |             dma_tx_ch->CMAR = (uint32_t)&tx_buf->data[tx_buf->tail];
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from usb_cdc.c:10:
circ_buf.h:15:13: note: while referencing 'data'
   15 |     uint8_t data[0];
      |             ^~~~
usb_cdc.c: In function 'DMA1_Channel4_IRQHandler':
usb_cdc.c:447:41: warning: array subscript '<unknown>' is outside the bounds of an interior zero-length array 'uint8_t[0]' {aka 'unsigned char[]'} [-Wzero-length-bounds]
  447 |             dma_tx_ch->CMAR = (uint32_t)&tx_buf->data[tx_buf->tail];
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from usb_cdc.c:10:
circ_buf.h:15:13: note: while referencing 'data'
   15 |     uint8_t data[0];
      |             ^~~~
usb_cdc.c: In function 'DMA1_Channel7_IRQHandler':
usb_cdc.c:447:41: warning: array subscript '<unknown>' is outside the bounds of an interior zero-length array 'uint8_t[0]' {aka 'unsigned char[]'} [-Wzero-length-bounds]
  447 |             dma_tx_ch->CMAR = (uint32_t)&tx_buf->data[tx_buf->tail];
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from usb_cdc.c:10:
circ_buf.h:15:13: note: while referencing 'data'
   15 |     uint8_t data[0];
      |             ^~~~
usb_cdc.c: In function 'DMA1_Channel2_IRQHandler':
usb_cdc.c:447:41: warning: array subscript '<unknown>' is outside the bounds of an interior zero-length array 'uint8_t[0]' {aka 'unsigned char[]'} [-Wzero-length-bounds]
  447 |             dma_tx_ch->CMAR = (uint32_t)&tx_buf->data[tx_buf->tail];
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from usb_cdc.c:10:
circ_buf.h:15:13: note: while referencing 'data'
   15 |     uint8_t data[0];
      |             ^~~~
usb_cdc.c: In function 'usb_cdc_poll':
usb_cdc.c:331:36: warning: array subscript '<unknown>' is outside the bounds of an interior zero-length array 'uint8_t[0]' {aka 'unsigned char[]'} [-Wzero-length-bounds]
  331 |                 uint8_t *buf_ptr = &rx_buf->data[rx_buf->tail];
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from usb_cdc.c:10:
circ_buf.h:15:13: note: while referencing 'data'
   15 |     uint8_t data[0];
      |             ^~~~

```

![Screenshot from 2021-07-13 18-35-35](https://user-images.githubusercontent.com/4820969/125481507-cb213f01-d943-4be7-b490-f0a94a2a305b.png)


Maybe exists version of gcc-arm-none-eabi which works fine w/o these warnings?

--- a/circ_buf.h
+++ b/circ_buf.h
@@ -12,7 +12,7 @@
 typedef struct  {
     int head;
     int tail;
-    uint8_t data[0];
+    uint8_t data[];
 } circ_buf_t;
 
 /* Returns number of bytes in buffer ()*/

